norse/dataset/yingyang/__init__.py

```diff
@@ -0,0 +1,118 @@
+# MIT License
+# 
+# Copyright (c) 2020 lkriener
+# 
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+# 
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+# 
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# For more information see https://github.com/lkriener/yin_yang_data_set
+
+from torch.utils.data.dataset import Dataset
+import numpy as np
+        
+class YinYangDataset(Dataset):
+    def __init__(self, r_small=0.1, r_big=0.5, size=1000, seed=42, transform=None):
+        super(YinYangDataset, self).__init__()
+        """
+Initializing the dataset:
+
+.. code: python
+    from norse.dataset.yingyang import YinYangDataset
+
+    dataset_train = YinYangDataset(size=5000, seed=42)
+    dataset_validation = YinYangDataset(size=1000, seed=41)
+    dataset_test = YinYangDataset(size=1000, seed=40)
+
+**Note** It is very important to give different seeds for trainings-, validation- and test set, as the data is generated randomly using rejection sampling. Therefore giving the same seed value will result in having the same samples in the different datasets!
```

Comment:
An approach to the random seed problem I've seen several times would be to leave the seed unset (default param `None`) to default to random values. In that way ignorant/busy users won't accidentally reuse the seed.
I think I would prefer that approach to trusting users to read the docs (yeah, right) and correctly set the parameter.

Reply:
Yeah that makes sense :).